pkg/curatedpackages/packagecontrollerclient.go

```diff
@@ -119,34 +123,64 @@ func (pc *PackageControllerClient) EnableCuratedPackages(ctx context.Context) er
 		noProxy := fmt.Sprintf("proxy.NO_PROXY=%s", strings.Join(pc.noProxy, "\\,"))
 		values = append(values, httpProxy, httpsProxy, noProxy)
 	}
-	if pc.eksaSecretAccessKey == "" || pc.eksaAccessKeyID == "" {
+	if (pc.eksaSecretAccessKey == "" || pc.eksaAccessKeyID == "") && pc.registryMirror == nil {
 		values = append(values, "cronjob.suspend=true")
 	}
 
-	err := pc.chartInstaller.InstallChart(ctx, pc.chart.Name, ociURI, pc.chart.Tag(), pc.kubeConfig, "", values)
-	if err != nil {
+	var err error
+	var valueFilePath string
+	if valueFilePath, _, err = pc.CreateHelmOverrideValuesYaml(); err != nil {
 		return err
 	}
 
-	// Customers are currently requesting to show a warning in case
-	// credentials are not provided for curated packages
-	if err = pc.CreateCredentials(ctx); err != nil {
-		logger.MarkWarning("  Unable to create credentials for curated packages: ", "warning", err)
+	if err := pc.chartInstaller.InstallChart(ctx, pc.chart.Name, ociURI, pc.chart.Tag(), pc.kubeConfig, "", valueFilePath, values); err != nil {
+		return err
 	}
 
 	return pc.waitForActiveBundle(ctx)
 }
 
-// CreateCredentials creates necessary credentials to enable the installation of curated packages.
-// In order to create the credentials, there are two necessary steps:
-//   - Creating a kubernetes secret
-//   - Creating a single run of the cron job to ensure the secret is consumed
-func (pc *PackageControllerClient) CreateCredentials(ctx context.Context) error {
-	if err := pc.ApplySecret(ctx); err != nil {
-		return errors.New("unable to detect AWS authentication credentials for curated packages. Skipping package installation. Follow documentation to set credentials after cluster creation completes")
+// CreateHelmOverrideValuesYaml creates a temp file to override certain values in package controller helm install.
+func (pc *PackageControllerClient) CreateHelmOverrideValuesYaml() (string, []byte, error) {
+	content, err := pc.generateHelmOverrideValues()
+	if err != nil {
+		return "", nil, err
 	}
+	if pc.valuesFileWriter == nil {
+		return "", content, fmt.Errorf("valuesFileWriter is nil")
+	}
+	filePath, err := pc.valuesFileWriter.Write(valueFileName, content)
+	if err != nil {
+		return "", content, err
+	}
+	return filePath, content, nil
+}
 
-	return nil
+func (pc *PackageControllerClient) generateHelmOverrideValues() ([]byte, error) {
+	var err error
+	endpoint, username, password, caCertContent := "", "", "", ""
+	if pc.registryMirror != nil {
+		endpoint = pc.registryMirror.BaseRegistry
+		username, password, err = config.ReadCredentials()
```

Comment:
ideally we would inject the username/password to the constructor. I'm ok having that as a follow up.